packages/llamaindex/src/indices/vectorStore/index.ts

```diff
@@ -335,15 +336,28 @@ export class VectorStoreIndex extends BaseIndex<IndexDict> {
     if (!nodes || nodes.length === 0) {
       return;
     }
-    nodes = await this.getNodeEmbeddingResults(nodes, options);
-    await addNodesToVectorStores(
-      nodes,
-      this.vectorStores,
-      this.insertNodesToStore.bind(this),
-    );
+    try {
+      const embeddedNodes = await this.getNodeEmbeddingResults(nodes, options);
+      await addNodesToVectorStores(
+        embeddedNodes,
+        this.vectorStores,
+        this.insertNodesToStore.bind(this),
+      );
+    } catch (error) {
+      await this.cleanDocStore(nodes);
```

Comment:
don't understand that - if `getNodeEmbeddingResults` fails, we don't add any nodes to the store - why do we need to call `cleanDocStore` then?